conans/client/conf/detect.py

```diff
@@ -161,6 +164,55 @@ def _get_profile_compiler_version(compiler, version, output):
     return version
 
 
+def _detect_gcc_libcxx(executable, version, output, profile_name, profile_path):
+    # Assumes a working g++ executable
+    new_abi_available = Version(version) >= Version("5.1")
+    if not new_abi_available:
+        return "libstdc++"
+
+    if not get_env(CONAN_V2_MODE_ENVVAR, False):
+        msg = textwrap.dedent("""
+            Conan detected a GCC version > 5 but has adjusted the 'compiler.libcxx' setting to
+            'libstdc++' for backwards compatibility.
+            Your compiler is likely using the new CXX11 ABI by default (libstdc++11).
+
+            If you want Conan to use the new ABI for the {profile} profile, run:
+
+                $ conan profile update settings.compiler.libcxx=libstdc++11 {profile}
+
+            Or edit '{profile_path}' and set compiler.libcxx=libstdc++11
+            """.format(profile=profile_name, profile_path=profile_path))
+        output.writeln("\n************************* WARNING: GCC OLD ABI COMPATIBILITY "
+                       "***********************\n %s\n************************************"
+                       "************************************************\n\n\n" % msg,
+                       Color.BRIGHT_RED)
+        return "libstdc++"
+
+    main = textwrap.dedent("""
+        #include <string>
+
+        using namespace std;
+        static_assert(sizeof(std::string) != sizeof(void*), "using libstdc++");
```

Comment:
static_assert is C++11, can we use it?

Reply:
maybe we can just include <string> and check the value of `_GLIBCXX_USE_CXX11_ABI`?

Reply:
It is only applied for gcc>=5.1 which already support c++11 (passed in the command line)

> maybe we can just include and check the value of _GLIBCXX_USE_CXX11_ABI?

That is the original problem, as you described in https://github.com/conan-io/conan/issues/7264#issuecomment-651107547, this will be ignored. We need to check for the real thing.

Reply:
oh, I see, you're passing `-std=c++11` below